rest/functionsapitest/graphql_test.go

```diff
@@ -0,0 +1,863 @@
+//  Copyright 2022-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package functionsapitest
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/graphql-go/graphql"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/db/functions"
+	"github.com/couchbase/sync_gateway/rest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const kDummyGraphQLSchema = `
+    type Query {
+        square(n: Int!) : Int!
+    }`
+
+var allowAll = &functions.Allow{Channels: []string{"*"}}
+
+// The GraphQL schema:
+var kTestGraphQLSchema = `
```

Comment:
Much of this source file is copied and pasted from the functions package unit tests. That is not a good idea:
1. Testing the same functionality (just called through a different API) is redundant and doesn't improve coverage.
2. Even if you're using the config to test something different (like parameter marshaling), the use of the exact same config and logic in the functions guarantees that the functions package isn't doing anything different than in the unit tests.